python/cudf/cudf/tests/test_text.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+import pytest
+from pandas.util.testing import assert_series_equal
+
+import cudf
+
+
+def test_tokenize():
+    strings = cudf.Series(
+        [
+            "the quick fox jumped over the lazy dog",
+            "the siamésé cat jumped under the sofa",
+            None,
+            "",
+        ]
+    )
+
+    expected = cudf.Series(
+        [
+            "the",
+            "quick",
+            "fox",
+            "jumped",
+            "over",
+            "the",
+            "lazy",
+            "dog",
+            "the",
+            "siamésé",
+            "cat",
+            "jumped",
+            "under",
+            "the",
+            "sofa",
+        ]
+    )
+
+    actual = strings.str.tokenize()
+
+    assert_series_equal(expected.to_pandas(), actual.to_pandas())
+
+
+@pytest.mark.parametrize(
+    "delimiter, expected_token_counts",
+    [
+        (" ", [10, 9, 0, 0, 1]),  # TODO: verify last count should be 1, not 5
+        ("o", [6, 3, 0, 0, 1]),
+        (["a", "e", "i", "o", "u"], [13, 13, 0, 0, 6]),
+        (["a", "e", "i", "o"], [12, 11, 0, 0, 6]),
+        # ([], [10, 9, 0, 0, 5]), # throws
```

Comment:
@kkraus14 @davidwendt This test case throws due to an empty delimiter list, but the expected counts are the same as the first test case (default / `" "` delimiter). Should I fall back to the default delimiter if the list is empty, or is this intentional?

Reply:
This is an error introduced by the port. The original NVText C++ code was rerouting to use the `token_count()` with `delimiter=nullptr` function when an empty delimiter list was passed. The libcudf port is not doing this. You can fix this in the cython/python layer if you wish or I can fix it in the nvtext ported code (or both).

Reply:
I'd like to fix it on the python side first, and we can remove it later as part of the port fix.

Reply:
Actually, I suspect the current implementation is a better fit. The user is attempting to invoke a method which expects a non-empty list. re-routing to a non-list method with a default value is unexpected behavior from my point of view. What if we adjust the test case instead?

Reply:
Ok